pkg/settings/setting.go

```diff
@@ -285,6 +285,8 @@ func init() {
 }
 
 // Provider is an interfaced used to get and set Settings.
+// NOTE: The behavior for treating unknown settings is undefined.
+// A provider may choose not to retain and/or drop unknown settings.
```

Comment:
NIt: I would move this to the "SetAll" function, since this behavior is particular to that function (Set and Get don't exhibit this behavior and I think it would be unreasonable for them to do so). I would also rephrase to something like:

```suggestion
// NOTE: The behavior for treating unknown settings is undefined.
// A provider may choose to remove settings which it has a record of, but are not provided in a new call to SetAll.
```

Reply:
Thanks for better phrasing, but I'm keeping on the interface itself. The provider may choose not to have such behavior or implement it in a different manner (e.g. store the names of provided settings and have a goroutine that proactively checks and does the cleaning). In other words the interface shouldn't prescribe it.